pkgs/development/mobile/androidenv/default.nix

```diff
@@ -29,9 +29,15 @@ rec {
       "34"
       "35"
     ];
-    includeEmulator = true;
-    includeSystemImages = true;
-    includeNDK = true;
+    includeEmulator =
+      with pkgs.stdenv.hostPlatform;
+      system == "x86_64-linux" || system == "x86_64-darwin" || system == "aarch64-darwin";
+    includeSystemImages =
+      with pkgs.stdenv.hostPlatform;
+      system == "x86_64-linux" || system == "x86_64-darwin" || system == "aarch64-darwin";
+    includeNDK =
+      with pkgs.stdenv.hostPlatform;
+      system == "x86_64-linux" || system == "x86_64-darwin" || system == "aarch64-darwin";
```

Comment:
Maybe we should pass an "if-supported" string here. Then if it's "if-supported" we can just not include it instead of erroring.

Reply:
I think I have problem understanding what would be the implementation for "if-supported"! would you explain more? Thanks!

Reply:
If there's a NDK package or similar for the current arch, we link it in. Otherwise we just eval warn about it.